### ISSUE TITLE
Remove the name of this overqualified element for hover

### ISSUE THREAD

**Issue by Eschon** [SonarCommunity/sonar-css#136]
Here's a screenshot of the issue:
![screen shot 2015-09-18 at 14 14 42](https://cloud.githubusercontent.com/assets/11647830/9958972/a6d08040-5e0f-11e5-8d15-d04d7f9f891a.png)

Removing the selector before hover hardly ever makes sense as far as I know. So I think this Issue should be disabled for hover.

**Comment by racodond** [comment]
@Eschon Thanks again for your feedback!

> Removing the selector before hover hardly ever makes sense as far as I know. So I think this Issue should be disabled for hover.

Actually the issue is not around `hover`. The issue comes from the fact that `li:hover .skimap-control-slopes:after` is parsed as `li:hover.skimap-control-slopes:after` (without the whitespace after `hover`).

**Comment by kalidasya** [comment]
@racodond I remember that I fixed this issue before (not the hover one).. so it got broken again... maybe we need a test for that.

**Comment by kalidasya** [comment]
@racodond probably the pseudo is not : and ANY but : and IDENT but in this case without whitespace

**Comment by racodond** [comment]
> probably the pseudo is not : and ANY but : and IDENT but in this case without whitespace

Unfortunately, it is not that simple. See http://www.w3schools.com/cssref/css_selectors.asp

> I remember that I fixed this issue before (not the hover one).. so it got broken again... maybe we need a test for that

I added some tests to the rule.
I'll try and add some tests to the grammar.

**Comment by kalidasya** [comment]
so its either `ident` or `function` or `: and ident`

**Comment by racodond** [comment]
Actually, it's more like:
```
b.rule(PSEUDO).is(
      b.firstOf("::", ":"),
      b.firstOf(FUNCTION_PSEUDO, identNoWS)
      );
```

**Comment by kalidasya** [comment]
yes, that looks better

**Comment by racodond** [comment]
> I don't know if it is easily possible to test that a string matches a sequence of grammar rules. For example `.class .test` matches two CLASS_SELECTOR and not one single CLASS_SELECTOR. @kalidasya Any idea?

I had a talk with @dbolkensteyn today at the SonarQube Conference and it is not possible to test the structure of the AST.
@kalidasya Thus, I'll let you review https://github.com/SonarCommunity/sonar-css/pull/141 and merge if it's ok with you.

**Comment by Haeresis** [comment]
I think I have exactly the same issue so, in this case, the solution is to manually say « Resolve as won't fix » it's correct ?

![image](https://cloud.githubusercontent.com/assets/3882169/14012948/a15584c0-f1a9-11e5-9591-beb638c39202.png)

**Comment by racodond** [comment]
Hi @Haeresis,

This issue has been fixed in version 1.6 of the CSS plugin. I gave a try with your code sample and it does not trigger any issue with version 1.7 of the plugin. Which version of the plugin are you running? If you can reproduce the issue with version 1.7, could you please privately share the whole CSS file to me so that I can try to figure out the issue?

Thank you

**Comment by Haeresis** [comment]
I upload the latest version of pluging and that fix effectively my issue.

Thx a lot :+1:
